js/widgets.js

```diff
@@ -1,108 +1,172 @@
-Array.range = function(n) {
-    // Array.range(5) --> [0,1,2,3,4]
-    return Array.apply(null, Array(n)).map((x, i) => i);
-};
+/* This file is using http://standardjs.com/ */
+// `fetch` polyfill added by webpack.
+/* global fetch */
 
-Object.defineProperty(Array.prototype, 'chunk', {
-    // Chunking an Array given the numeric parameter.
-    value: function(n) {
-        return Array.range(Math.ceil(this.length / n)).map((x, i) => this.slice( i * n, i * n + n));
-    }
-});
+// Constants in use within the HTML.
+const TERRITORY = 'data-udata-territory'
+const TERRITORY_ID = 'data-udata-territory-id'
+const DATASET_ID = 'data-udata-dataset-id'
 
-function checkStatus(response) {
-    // `fetch` doesn't provide an error handling based on status code.
-    if (response.status >= 200 && response.status < 300) {
-        return response;
-    } else {
-        const error = new Error(response.statusText);
-        error.response = response;
-        throw error;
-    }
+// Helpers
+const qS = document.querySelector.bind(document)
+const qSA = document.querySelectorAll.bind(document)
+
+// Extract the base URL from the URL of that script.
+const parser = document.createElement('a')
+const scriptURI = qS('script#udata').src
+parser.href = scriptURI
+const baseURL = `${parser.protocol}//${parser.host}`
+
+function chunk (array, n) {
+  // Return the `array` chunked by `n`, useful to retrieve all datasets
```

Comment:
Until now, the convention used for JS doctring was:
```js
/**
 * Documentation
 */
function func(param) {}
```
We might need to really discuss those choices and stick with it in the futur.